src/entities/communitySbt.ts

```diff
@@ -189,15 +224,167 @@ class SBT {
+            }
+            return badgesResponse;
+        } catch (error) {
+          return [];
+        }
+    }
+
+    public async getTopTraderBadge(
+        subgraphUrl: string,
+        userId: string,
+        seasonId: number,
+        badgeType: string
+      ): Promise<BadgeResponse | undefined> {
+        if (!process.env.REACT_APP_SUBGRAPH_BADGES_URL) {
+            return undefined;
+          }
+        try {
+            const badgeQuery = `
+                query( $id: BigInt, $unit: String) {
+                    seasonUsers(first: 5, orderBy: $unit, orderDirection: desc) {
+                        id
+                        badges {
+                        id
+                        awardedTimestamp
+                        mintedTimestamp
+                        badgeType
+                        badgeName
+                        }
+                        totalNotionalTraded
+                        totalLiquidityProvided
+                    }
+                }
+            `;
+            const client = new ApolloClient({
+                cache: new InMemoryCache(),
+                link: new HttpLink({ uri: subgraphUrl, fetch })
+            })
+            const id = `${userId.toLowerCase()}#${seasonId}`;
+            const isNotional = badgeType.indexOf("trade") > -1;
+            const unit = isNotional ? "totalNotionalTraded" : "totalLiquidityProvided";
+            const data = await client.query({
+                query: gql(badgeQuery),
+                variables: {
+                    id: id,
+                    unit: unit
+                },
+            });
+    
+            if (!data.data.seasonUsers) {
+                return undefined;
+            }
+
+            for (const seasonUser of data.data.seasonUsers) {
+                if (seasonUser.id === userId.toLowerCase()) {
+                    const badge : BadgeResponse = {
+                        id: `${userId}#${seasonId}#${badgeType}`,
+                        badgeType: badgeType,
+                        badgeName: TOP_BADGES_VARIANT[badgeType],
+                        awardedTimestamp: isNotional ? seasonUser.notionalAwardedTimestamp : seasonUser.liquidityAwardedTimestamp,
+                        mintedTimestamp: isNotional ? seasonUser.notionalMintedTimestamp : seasonUser.liquidityMintedTimestamp,
+                    }
+                    return badge;
+                }
+            }
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    public async getNonProgramaticBadges(userId: string, nonProgramaticBadgesUrl: string) : Promise<Record<string, BadgeResponse>> {
+        let badgeResponssRecord : Record<string, BadgeResponse> = {};
+
+        const resp = await axios.get(nonProgramaticBadgesUrl + userId);
+        if (!resp.data){
+            return badgeResponssRecord ;
+        }
+
+        const badges: NonProgramaticBadgeResponse[] = resp.data.badges;
+        badges.map((entry) => {
```

Comment:
map returns an array. Since you really don't need the return of map you can replace it with .forEach
```suggestion
        badges.forEach((entry) => {
```